src/extension.ts

```diff
@@ -3723,6 +3723,160 @@ export async function activate(context: vscode.ExtensionContext) {
   );
 
   checkAndNotifyMissingCompileCommands();
+
+  // Remove ESP-IDF settings
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "espIdf.removeEspIdfSettings",
+      removeEspIdfSettings
+    )
+  );
+}
+
+async function removeEspIdfSettings() {
+  const config = vscode.workspace.getConfiguration();
+  const settingsToDelete: string[] = [];
+
+  // Helper function to recursively find idf settings
+  function findIdfSettings(obj: any, prefix: string = "") {
+    if (typeof obj === "object" && obj !== null) {
+      Object.keys(obj).forEach((key) => {
+        const fullPath = prefix ? `${prefix}.${key}` : key;
+        if (fullPath.startsWith("idf.") || fullPath.startsWith("esp.")) {
+          settingsToDelete.push(fullPath);
+        }
+        findIdfSettings(obj[key], fullPath);
+      });
+    }
+  }
+
+  // Get all settings directly from configuration
+  const allSettings = config.inspect("");
+  // Check values saved in each scope
+  if (allSettings?.globalValue) {
```

Comment:
findIdfSettings can be simplified into a for loop with an array [allSettings.globalValue, allSettings.workspaceValue, allSettings.workspaceFolderValue]